roles/kubespray-defaults/defaults/main.yaml

```diff
@@ -335,15 +335,16 @@ etcd_events_access_addresses: |-
   {% for item in groups['etcd'] -%}
     https://{{ hostvars[item]['access_ip'] | default(hostvars[item]['ip'] | default(hostvars[item]['ansible_default_ipv4']['address'])) }}:2381{% if not loop.last %},{% endif %}
   {%- endfor %}
+# user should set etcd_member_name in inventory/mycluster/hosts.ini
 etcd_member_name: |-
   {% for host in groups['etcd'] %}
-  {%   if inventory_hostname == host %}{{"etcd"+loop.index|string }}{% endif %}
+  {%   if inventory_hostname == host %}{{ hostvars[host].etcd_member_name }}{% endif %}
```

Comment:
there is no else clause, so if you don't add `etcd_member_name` in inventory you are screwed with empty etcd_member_name

Reply:
yes, so we must  add etcd_member_name in inventory. Please see the added annotations in inventory.
We solve the problem of loop.index at the cost of increasing etcd_member_name in inventory. That is , to restrict the user to use the condition.
I think loop.index is inappropriate. When the order is changed, fault is made. e.g. two examples described above.
Thanks!

Reply:
What should I do? Add else clause?

Reply:
A non etcd member does not have the opportunity to execute etcd_member_name.

Reply:
that if is not needed there if no else clause is provided, also if i'm upgrading cluster i need to add those? or it will work with old inventory too? for new installs ok people read doc (not always) but for old inventories people don't intend to read what's changed in release, so there should be some way to trigger error with message even before we get to do something to not fail on other things. I cannot think of way for now, as kubespray-defaults is first readed, but `if` will not help in this case

basicly what i want is to fail if requirements are not met as this is very essential part of logic later.

Reply:
I see. I will change it. Thanks!